lib/zinnia.js

```diff
@@ -91,42 +119,47 @@ export async function run ({
   ])
 }
 
-function handleEvents ({ onActivity, onMetrics, text }) {
+async function handleEvents ({
+  contract,
+  FIL_WALLET_ADDRESS,
+  onActivity,
+  onMetrics,
   text
-    .trimEnd()
-    .split(/\n/g)
-    .forEach(line => {
-      try {
-        const event = JSON.parse(line)
-        switch (event.type) {
-          case 'activity:info':
-            onActivity({
-              type: 'info',
-              message: event.message.replace(/Module Runtime/, 'Zinnia'),
-              source: event.module
-            })
-            break
+}) {
+  for (const line of text.trimEnd().split(/\n/g)) {
+    try {
+      const event = JSON.parse(line)
+      switch (event.type) {
+        case 'activity:info':
+          onActivity({
+            type: 'info',
+            message: event.message.replace(/Module Runtime/, 'Zinnia'),
+            source: event.module
+          })
+          break
 
-          case 'activity:error':
-            onActivity({
-              type: 'error',
-              message: event.message.replace(/Module Runtime/, 'Zinnia'),
-              source: event.module
-            })
-            break
+        case 'activity:error':
+          onActivity({
+            type: 'error',
+            message: event.message.replace(/Module Runtime/, 'Zinnia'),
+            source: event.module
+          })
+          break
 
-          case 'jobs-completed':
-            onMetrics({
-              totalJobsCompleted: event.total,
-              totalEarnings: '0'
-            })
-            break
+        case 'jobs-completed':
+          onMetrics({
+            totalJobsCompleted: event.total,
+            rewardsScheduledForAddress:
+              (await contract.rewardsScheduledFor(FIL_WALLET_ADDRESS))
+                .toString()
```

Comment:
IIUC, if we cannot fetch the rewards, then the entire `handleEvents` call fails with an error. I think this is brittle: if `text` contains multiple events, the first one is `jobs-completed`, and the contract query fails, then we end up not processing all of the events.

Proposal:
- Extract a new function to query the scheduled rewards
- Handle RPC errors inside this function

Reply:
Good call! Addressed in https://github.com/filecoin-station/core/pull/267